hy/lex/parser.py

```diff
@@ -50,8 +60,8 @@ def mangle(s):
                  # allowed at the start of an identifier.
                else '{0}{1}{0}'.format(mangle_delim,
                    unicodedata.name(c, '').lower().replace('-', 'H').replace(' ', '_')
-                   or 'U{:x}'.format(ord(c)))
-            for c in s)
+                   or 'U{}'.format(c.encode('unicode-escape').decode('utf-8').lstrip('\\U').lstrip('\\u').lstrip('0')))
```

Comment:
So long as `c` is unicode, `ord` should work:

```python
or 'U{:x}'.format(ord(c))
```

Reply:
If c is a two 16-bit int unicode, `ord(c)` throws in a UCS2 python system:

```
>>> ord(u'\U0001f5b1')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: ord() expected a character, but string of length 2 found
```

But I have to say my solution is really ugly - just a patch I can think of without refactor the function.

Reply:
Oh of course it be the same problem.

Reply:
And I just noticed the old code did ord. Okay. So I just have one more suggestion to make and I'm happy with this:

I have a feeling someone is going to come along, look at this, and say someone "Wait... isn't this ord?" and break this all over again. Not to mention we have to remember to pull it out once we finally drop Python 2 support.

So may I suggest adding a `multibyte_ord` helper?

```
def multibyte_ord(ustr):
    if UCS4:
        return ord(ustr)
    return ustr.encode('unicode-escape').decode('utf-8').lstrip('\\U').lstrip('\\u').lstrip('0')
```

Makes it clearer that this klugde is in there intentionally?

Reply:
That makes sense to me.